Progress Report - September 2022 (#334)
yuzu-emu/yuzu-emu.github.io: site/content/entry/yuzu-progress-report-sep-2022/index.md

Nintendo Online. The paid Nintendo service to play online, which also includes other bonuses as official emulators to play older titles. As many know, we have some 
history with this method. At least while the console is in service, we won’t pursue offering support for Nintendo Online.

`Project London` includes full support for LDN, with room hosting and joining, and revamping LAN support to use the new LDN code in place.

So, how to use it.

The user can join the room lobby from either the Multiplayer menu up top selecting the *Browse Public Game Lobby*, or by clicking the bottom right status bar tip *Not Connected. Click here to find a room!*

{{< imgs
	"./ldn.png| The new Multiplyaer submenu"
  >}}

From there, double clicking a room will allow the user to join them. Keep in mind the Preferred Game is just a suggestion, there is no restriction on which games can be played in a room, unless an user hosted one wants to enforce it.

{{< imgs
	"./rooms.png| A lock means it's a password protected room"
  >}}

> Speaking of user hosted rooms, the *Create Room* option in the Multiplayer menu allows users to host their own rooms, with custom player limits and password ecurity.

Reviewer: ```suggestion
Speaking of user hosted rooms, the *Create Room* option in the Multiplayer menu allows users to host their own rooms, with custom player limits and password security.
```

And that’s it! Join rooms, both in yuzu and in game, and fight! Or trade those Pokémon, your call.
This endeavour was possible thanks to the [ldn_mitm project](https://github.com/spacemeowx2/ldn_mitm) who provided us a licence exemption, and the work done by [Tobi](https://github.com/FearlessTobi) and [german77](https://github.com/german77), along with the testers (your writer included), who never had so much fun crashing games in the name of progress.

After the Mainline release, Linux users reported to us that LDN rooms refused to work on some Linux distributions.
The issue is in how the SSL package is distributed, so  [toastUnlimited](https://github.com/lat9nq), taking reference from a previous pull request from [Docteh](https://github.com/Docteh) {{< gh-hovercard "8933" " fixed the issue" >}} so now all supported operating systems can enjoy LDN.

Some users reported crashes when playing `Pokémon Sword/Shield`. 
While we implement the correction for it, make sure that a valid network device is selected in `Emulation > Configure… > System > Network > Network Interface`.
The crash seems to happen when “None” is set as the network interface.

## Graphic changes

> Nvidia OpenGL users OpenGL may have seen some strange graphical glitches in games like `Dragon Quest Builder`, `SNACK WORLD: THE DUNGEON CRAWL`, and `Pixel Game Maker Series Werewolf Princess Kaguya` due to an oversight in yuzu's handling of the performance-enhancing `vertex_buffer_unified_memory` extension. 

Reviewer: ```suggestion
Nvidia OpenGL users may have seen some strange graphical glitches in games like `Dragon Quest Builder`, `SNACK WORLD: THE DUNGEON CRAWL`, and `Pixel Game Maker Series Werewolf Princess Kaguya` due to an oversight in yuzu's handling of the performance-enhancing `vertex_buffer_unified_memory` extension. 
```

The old SD card icon has been {{< gh-hovercard "8902" "replaced with a new colourful microSD Card icon" >}} by [Dev-draco](https://github.com/Dev-draco).

{{< imgs
	"./sdicon.png| Small details matter"
  >}}

[Tachi107](https://github.com/Tachi107) {{< gh-hovercard "8945" "fixed a few minor typos" >}} within the yuzu source code. 

german77 modified the input profiles He also fixed the {{< gh-hovercard "8948" "sorting to now sort them by name." >}}

While the colourful theme has been the default for a while, some icons remained in a "Default" theme directory.
But as grabbing icons from other themes isn't supported on linux, this was broken.
Docteh fixed this by {{< gh-hovercard "8906" "moving all the icons to the colourful theme." >}}

Since the **Debug** configuration tab was getting crowded, Docteh {{< gh-hovercard "8854" "made it scrollable." >}}

{{< single-title-imgs
    "While this change helps Windows, Linux users will benefit the most of it"
    "./scrollbug.png"
    "./scrollfix.png"
    >}}

Docteh also fixed the {{< gh-hovercard "8909" "broken help page hyperlink on the `TAS` configuration window" >}}.

{{< imgs

> 	"./tas.png| Poeple seem to forget that blue underlined text is an hyperlink intended to be clicked"

Reviewer: ```suggestion
	"./tas.png| People seem to forget that blue underlined text is an hyperlink intended to be clicked"
```

But since this feature has a big performance impact, this was added behind an option in the **Debug** settings for users to access.
You can find it in `Emulation > Configure… > General > Debug > Create Minidump After Crash`.

{{< imgs
	"./minidump.png| Only use it if you want to debug a game or to pass the file to a developer, the performance loss is big!"
  >}}

Since the dumps use modern Windows SDK features, there’s a risk that this change can break compatibility with older Windows options.
yuzu only officially supports Windows versions 10 1803 and newer. 
And Linux, of course. It's faster than Windows.

## Hardware section

Users started reporting crashes when booting up `Xenoblade Chronicles 3` on AMD Radeon GPUs running official AMD drivers (either the Windows driver, or the Linux amdvlk driver).
We’re investigating and already passed AMD the relevant information.

Affected users can try running [Mainline 1188](https://github.com/yuzu-emu/yuzu-mainline/releases/tag/mainline-0-1188) (doesn’t seem to be a consistent fix), or just run the game using OpenGL.
The Mesa RADV driver is unaffected.

> 

Reviewer: Maybe a note saying  {{< gh-hovercard "8849" "astc: Enable parallel CPU astc decoding " >}} is merged now as this happened 3 days after the previous report was published and Maxwell and Pascal GPUs can upgrade their drivers, or if not, what the current status is for those GPUs and Vulkan.

Author: Thanks, added.

Affected users can try running [Mainline 1188](https://github.com/yuzu-emu/yuzu-mainline/releases/tag/mainline-0-1188), or just run the game using OpenGL.
The Mesa RADV driver is unaffected.

#### NVIDIA, one crash reason down, more remain

We have merged the [multithreaded ASTC CPU decoder](https://github.com/yuzu-emu/yuzu/pull/8849) mentioned last month, so we strongly recommend NVIDIA Maxwell and Pascal (GTX 745/750, 900 and 1000 series) users to disable ASTC decoding if you run the latest drivers.

You can find the option in `Emulation > Configure... > Graphics > Accelerate ASTC Texture Decoding`, an enabled setting uses the GPU to decode, while a disabled one uses the CPU instead.
As a bonus, now this option can be a performance boost on systems with very weak integrated GPUs, like old laptops with MX series Geforce hardware, or any Intel iGPU, where the CPU would do a faster job dealing with ASTC decoding. Testing has shown that MX 500 and > Vega 3 tier hardware are slightly faster using the GPU for decoding.

> On affected NVIDIA hardware, the performance decrease while decoding ASTC textures is minimal. Avoiding a crash for example when opening the map in `The Legend of Zelda` games is arguably far more important.

Author: I mean bigger than 3.
Vega 3 is the model with 3 compute units, very slow. The next models, Vega 6 and higher are strong enough.

No news yet on Vulkan support for these new AV1 video decoder cards from Intel that can also work as a GPU. 
Low avaibility has been a big problem to get our hands on them, but it's something we want to solve ASAP.

OpenGL is "functional" right now, but Intel continues to be the worst vendor regarding OpenGL support on Windows, so the experience is pretty bad.
Linux Mesa drivers are reported to work correctly and with good performance, in both OpenGL and Vulkan.

Speaking of AV1... For now we've been using h.264 encoding for our videos, as it is both widely supported and very light on performance requirements.
Would you prefer if we switched to AV1 from now on? It can be a big battery drainer on unsupported hardware, but it would allow for even lower bitrates while keeping a similar level of quality, something datacapped users have complained about in the past.
Let us know in Reddit, our [forums](https://community.citra-emu.org/c/yuzu-news/15) or our [Discord server](https://discord.gg/u77vRWY) what you would prefer!

## Future projects

For once, we don’t have much to say in this section.
Or at least, we don’t want to start generating hype about new projects just yet.

> [Blinkhawk](https://github.com/FernandoS27) has started working on part 2 of `Project Y.F.C.`, as previously mentioned, and the rest of the GPU gang (Byte[], Maide, vonchenplus) is also busy with some top secret changes (for now, your writer has a reputation to uphold) to improve both rendering accuracy and performance.
> 
> Having the first part of Y.F.C. merged to Mainline opens the floodgates to many projects that were on hold until now, we will talk about more of them in the next progress report.
> 
> That’s all folks! As always, many thanks for your time. We hope to see you next.
> 
> Special thanks to Mysterious Writer B for their help, and to CaptV0rt3x who came to aid at a moment’s notice.

Reviewer 2: ```suggestion
[Blinkhawk](https://github.com/FernandoS27) has started working on Part 2 of `Project Y.F.C.`, as previously mentioned, and the rest of the GPU gang (Byte[], Maide, vonchenplus) is also busy with some top secret changes (for now, your writer has a reputation to uphold) to improve both rendering accuracy and performance.

Having the first part of `Y.F.C.` merged to Mainline opens the floodgates to many projects that were on hold until now. We will discuss these in more detail in the next progress report.

Special thanks to Mysterious Writer B for their help with this progress report, and to CaptV0rt3x who came to aid at a moment’s notice.

That’s all folks! As always, many thanks for your time. We hope to see you in the next article!
```
